nucliadb/nucliadb/ingest/orm/processor/__init__.py

```diff
@@ -605,36 +407,50 @@ async def apply_resource(
 
         return None
 
-    async def notify_commit(
-        self, partition: str, seqid: int, multi: str, kbid: str, uuid: str
+    def notify_commit(
+        self,
+        *,
+        partition: str,
+        seqid: int,
+        multi: str,
+        message: BrokerMessage,
+        write_type: Notification.WriteType.Value,  # type: ignore
     ):
-        message = Notification(
+        notification = Notification(
             partition=int(partition),
             seqid=seqid,
             multi=multi,
-            uuid=uuid,
-            kbid=kbid,
-            action=Notification.COMMIT,
+            uuid=message.uuid,
+            kbid=message.kbid,
+            action=Notification.Action.COMMIT,
+            # including the message here again might feel a bit unusual but allows
+            # us to react to these notifications with the original payload
+            write_type=write_type,
+            message=message,
         )
-        await self.notify(
-            const.PubSubChannels.RESOURCE_NOTIFY.format(kbid=kbid),
-            message.SerializeToString(),
+        asyncio.create_task(
+            self.notify(
+                const.PubSubChannels.RESOURCE_NOTIFY.format(kbid=message.kbid),
+                notification.SerializeToString(),
+            )
         )
 
-    async def notify_abort(
-        self, partition: str, seqid: int, multi: str, kbid: str, uuid: str
+    def notify_abort(
+        self, *, partition: str, seqid: int, multi: str, kbid: str, rid: str
```

Comment:
why is this `*`, useful for?

Reply:
it forces the calling to provide all keyword args. I can remove it if you like but I prefer to have all named arguments for readability and preventing position argument errors.

Reply:
nice! no, I like it. I was wondering for a long time... where can I read about this in the python docs?